tools/tiller-hack/tiller-hack.go

```diff
@@ -0,0 +1,256 @@
+// tracking of CRDs objects by helm's tiller after they were deployed by means of
+// helm's charts.
+//
+// ./tiller-hack --kubeconfig={location of kube config file} \
+//               --configmap={the name of tiller's config map for istio deployment}
+//               --namespace={the namespace of configmap and tiller,
+//                            by default it is kube-system}
+//               --restart-tiller={true/false, false is default, true will
+//                                 force the tool to restart tiller}
+//
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/golang/protobuf/proto"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+	rspb "k8s.io/helm/pkg/proto/hapi/release"
+)
+
+var (
+	kubeconfig = flag.String("kubeconfig", "",
+		"Absolute path to the kubeconfig file, must be specified.")
+	configMapName = flag.String("configmap", "",
+		"Name of the configmap to patch, must be specified.")
+	configMapNamespace = flag.String("namespace", "kube-system",
+		"Namespace of the configmap, defaults to \"kube-system\"")
+	restartTillerFlag = flag.Bool("restart-tiller", false,
+		"Restart tiller after patching the configmap")
+
+	// In older versions of helm, the compression of release information has not been used.
+	// This header identify whether the following info is compressed or not.
+	magicGzip = []byte{0x1f, 0x8b, 0x08}
+)
+
+// BuildClient returns kubernetes clientset
+func BuildClient(kubeconfig string) (*kubernetes.Clientset, error) {
+	var config *rest.Config
+	var err error
+
+	kubeconfigEnv := os.Getenv("KUBECONFIG")
+
+	if kubeconfigEnv != "" {
+		kubeconfig = kubeconfigEnv
+	}
+
+	if kubeconfig != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return nil, err
+	}
+	k8s, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return k8s, nil
+}
+
+func getConfigmapData(k8sClient *kubernetes.Clientset, configMapName, configMapNamespace string) ([]byte, error) {
+	cfgMap, err := k8sClient.Core().ConfigMaps(configMapNamespace).Get(configMapName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("fail to get configmap %s/%s with error: %+v", configMapNamespace, configMapName, err)
+	}
+	data, ok := cfgMap.Data["release"]
+	if !ok {
+		return nil, fmt.Errorf("fail to locate \"release\" key in configmap %s/%s", configMapNamespace, configMapName)
+	}
+	return []byte(data), nil
+}
+
+func makeConfigMapBackup(k8sClient *kubernetes.Clientset, configMapName, configMapNamespace string) error {
```

Comment:
this inevitably raises the question of how does one restore a configmap, especially with a timestamp.

Reply:
Easy, dump the content of backup configmap into a yaml file, then delete patched configmap and recreate the original configmap.